File: 1001-2000/1578-minimum-time-to-make-rope-colorful/1578-minimum-time-to-make-rope-colorful.cpp
class Solution {
public:
    int minCost(string colors, vector<int>& neededTime) {
        int min_cost=0;
        char prev = colors[0];
        int prev_index=0;
        for(int i=1; i<colors.length(); i++){
            if(colors[i]==prev){
                min_cost+=min(neededTime[prev_index],neededTime[i]);
                if(neededTime[prev_index]<=neededTime[i]){
                    prev=colors[i];
                    prev_index=i;
                }
            }
            else{
                prev=colors[i];
                prev_index=i;
            }
        }
        return min_cost;
    }
};

File: 1001-2000/1578-minimum-time-to-make-rope-colorful/1578-minimum-time-to-make-rope-colorful.java
class Solution {
    public int minCost(String colors, int[] neededTime) {
        //int n=neededTime.length;
        char[] str=colors.toCharArray();

        int i=0,j=1;
        int n=str.length;
        int time=0;

        while(j<n){
            if(str[i]==str[j]){
                if(neededTime[i]<neededTime[j]){
                    time+=neededTime[i];
                    i=j;
                    j++;
                }
                else{
                    time+=neededTime[j];
                    j++;
                }
            }
            else{
                i=j;
                j++;
            }
        }
        return time;

        
    }
}